### PR TITLE
Quiet file open errors during H5Fis_accessible

### PR DESCRIPTION
**Pull request by mattjala**
Fix for #59

### CODE REVIEW

File: src/rest_vol_file.c [hunk 1]
```diff
@@ -616,11 +617,19 @@ RV_file_specific(void *obj, H5VL_file_specific_args_t *args, hid_t dxpl_id, void
             /* Initialize in case of failure */
             *ret_is_accessible = FALSE;
 
-            void *ret_file;
+            H5E_BEGIN_TRY
+            {
+                file_id = H5Fopen(filename, H5F_ACC_RDWR, fapl_id);
```
Comment (jhendersonHDF) [inline src/rest_vol_file.c:622]: Why the change to H5Fopen instead of RV_file_open? In general it's best to try to stick with minimal calling of public HDF5 APIs inside VOL connectors where possible, although that can be difficult at times.

Comment (mattjala) [reply src/rest_vol_file.c:622]: I thought this would be more consistent with other places in the VOL, but after checking I realize that this isn't the case. Is there a reason that it's better to avoid use of the external API? I would expect it to be better practice to use it, since internal VOL functions often depend on things the library has set up.

Comment (jhendersonHDF) [reply src/rest_vol_file.c]: Calling public API functions can currently mess with HDF5's API contexts, error stacks, and other things. But yes it's not always possible to avoid, especially for things like needing to call H5T routines.

**Review by jhendersonHDF — APPROVED** [review]
